File: app/src/main/java/com/giphy/browser/common/network/Service.java
package com.giphy.browser.common.network;

import com.giphy.browser.common.model.Gifs;

import io.reactivex.Single;
import retrofit2.adapter.rxjava2.Result;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface Service {
    @GET("/v1/gifs/trending")
    Single<Result<Gifs>> getTrendingGifs(@Query("api_key") String apiKey, @Query("offset") int offset);

    @GET("/v1/gifs/search")
    Single<Result<Gifs>> getSearchGifs(@Query("api_key") String apiKey, @Query("q") String query, @Query("offset") int offset);
}

File: app/src/main/java/com/giphy/browser/common/network/Service.kt
package com.giphy.browser.common.network

import com.giphy.browser.common.model.Gifs
import io.reactivex.Single
import retrofit2.adapter.rxjava2.Result
import retrofit2.http.GET
import retrofit2.http.Query

interface Service {
    @GET("/v1/gifs/trending")
    fun getTrendingGifs(
        @Query("api_key") apiKey: String?,
        @Query("offset") offset: Int
    ): Single<Result<Gifs>>

    @GET("/v1/gifs/search")
    fun getSearchGifs(
        @Query("api_key") apiKey: String?,
        @Query("q") query: String?,
        @Query("offset") offset: Int
    ): Single<Result<Gifs>>
}
